python/ray/train/batch_predictor.py

```diff
@@ -106,6 +124,19 @@ def predict(
         ):
             predictor_kwargs["use_gpu"] = True
 
+        if feature_columns:
+            dropped_dataset = data.map_batches(
+                lambda df: df[feature_columns], batch_size=batch_size
+            )
+        else:
+            dropped_dataset = data
+
+        original_col_ds = None
+        if keep_columns:
+            original_col_ds = data.map_batches(
```

Comment:
i assume the cost of `.map_batches` is O(data_size), is there any API in dataset that takes both `feature_columns` and `keep_columns` so we have both values we need in one `.map_batches` pass ?

Reply:
For this I believe we need an API to output 2 datasets as the result of a map on an initial dataset.

There's no API to do this with Datasets right now.